### ISSUE TITLE
Image Border clipping incorrect on Mac/iOS

### ISSUE THREAD

**Issue by jknaudt21** [dotnet/maui#17560]
### Description

Mac counterpart of #17523

https://github.com/dotnet/maui/assets/32918747/8d51ec4f-1cb0-45e6-84ae-7a9c80067908

### Steps to Reproduce

Use the sample app "Border resize content" page

### Version with bug

8.0.0-rc.1.9171

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, macOS

**Comment by jstedfast** [comment]
@jknaudt21 yes, it's the same on Android - it's related to the Scale transform (and likely other transforms will also have issues)

**Comment by jknaudt21** [comment]
Dropping this issue as my work is being redirected outside of Maui. From what I started to investigate, this issue stems from the fact that the `ContentView` is:

- Not adjusting the image's center correctly. [Right now the mask is shifting around with the transform's applied to the mask](https://github.com/dotnet/maui/blob/main/src/Core/src/Platform/iOS/ContentView.cs#L113). A bandaid fix is to force it to be centered based on the Frame. Something like `adjustedMaskPosition  = new(content.Frame.GetMidX(), content.Frame.GetMidY()` 
- The mask is also resizing based on the transforms the image received. It shouldn't do that and instead transform to what the border was received. A bandaid fix is to do `_contentMask.Transform = _contentMask.Transform.Concat(scaleTransform.Invert())`, where the `scaleTransform` is just the M11, M22, and M33 positions of the transform matrix.

**Comment by XamlTest** [comment]
Verified this on VS 17.10.0 Preview 2.0(8.0.7). Repro on iOS 17.2 and MacCatalyst with below Project:
[BorderResizeContent.zip](https://github.com/dotnet/maui/files/14611602/BorderResizeContent.zip)
